src/apps/metastreet/ethereum/metastreet.lending-v2.contract-position-fetcher.ts

```diff
@@ -0,0 +1,311 @@
+          shares
+          tick {
+            id
+            value
+            shares
+          }
+          redemptions {
+            shares
+          }
+        }
+      }
+    }
+  `;
+}
+
+export type GetPoolsResponse = {
+  pools: {
+    id: string;
+    ticks: {
+      id: string;
+      limit: BigNumber;
+      duration: BigNumber;
+      rate: BigNumber;
+      raw: BigNumber;
+    }[];
+    currencyToken: string;
+    collateralToken: string;
+  }[];
+};
+
+export type GetAllDepositsResponse = {
+  pools: {
+    deposits: {
+      shares: BigNumber;
+      tick: {
+        id: string;
+        shares: BigNumber;
+        value: BigNumber;
+      };
+      redemptions: {
+        shares: BigNumber;
+      }[];
+    }[];
+  }[];
+};
+
+export type GetDepositsResponse = {
+  pool: {
+    deposits: {
+      shares: BigNumber;
+      tick: {
+        id: string;
+        shares: BigNumber;
+        value: BigNumber;
+      };
+      redemptions: {
+        shares: BigNumber;
+      }[];
+    }[];
+  };
+};
+
+export const SUBGRAPH_URL = 'https://api.thegraph.com/subgraphs/name/metastreet-labs/metastreet-v2-beta';
+
+export type ContractPositionDefinition = {
+  address: string;
+  tickId: string;
+  tick: BigNumber;
+  limit: BigNumber;
+  duration: BigNumber;
+  rate: BigNumber;
+  currencyTokenSymbol: string;
+  collateralTokenName: string;
+};
+
+export type DataProps = {
+  positionKey: string; // tick id
+  tick: BigNumber; // encoded tick
+};
+
+@PositionTemplate()
+export class EthereumMetastreetLendingV2ContractPositionFetcher extends ContractPositionTemplatePositionFetcher<Pool> {
+  groupLabel: string = 'Lending';
+
+  constructor(
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+    @Inject(MetastreetContractFactory) protected readonly metastreetContractFactory: MetastreetContractFactory,
+  ) {
+    super(appToolkit);
+  }
+
+  getContract(_address: string): Pool {
+    return this.metastreetContractFactory.pool({ address: _address, network: this.network });
+  }
+
+  async getDefinitions(_params: GetDefinitionsParams): Promise<ContractPositionDefinition[]> {
+    const data = await gqlFetch<GetPoolsResponse>({
+      endpoint: SUBGRAPH_URL,
+      query: GET_POOLS_QUERY,
+    });
+
+    return data.pools.flatMap(p => {
+      return p.ticks.map(t => ({
+        address: p.id,
+        tickId: t.id,
+        tick: t.raw,
+        limit: t.limit,
+        duration: t.duration,
+        rate: t.rate,
+        currencyTokenSymbol: p.currencyToken['symbol'],
+        collateralTokenName: p.collateralToken['name'],
+      }));
+    });
+  }
+
+  async getTokenDefinitions(
+    _params: GetTokenDefinitionsParams<Pool, ContractPositionDefinition>,
+  ): Promise<UnderlyingTokenDefinition[] | null> {
+    const pool: Pool = _params.contract;
+    return [
+      {
+        metaType: MetaType.CLAIMABLE,
+        address: await pool.currencyToken(),
+        network: this.network,
+      },
```

Comment:
This position only has a claimable position and you don't need to supply any token ?

Reply:
Yes, users do need to supply a token. We aren't sure how the token balances are needed on Zapper's frontend and hence, was sure if it is needed. 

We will go ahead and add that, and will update in this conversation when that is done

Reply:
We added a `SUPPLIED` token